### ISSUE TITLE
[BUG] Non-empty null rows encountered with `subquery_test.py::test_scalar_subquery_array`

### ISSUE THREAD

**Issue by mythrocks** [NVIDIA/spark-rapids#11090]
Found as part of https://github.com/NVIDIA/spark-rapids/issues/11029.

When `subquery_test.py::test_scalar_subquery_array` is run with `spark.sql.ansi.enabled=true`, it unearths a CUDF column that contains non-empty null rows:
```
E                       Suppressed: com.nvidia.spark.rapids.jni.GpuRetryOOM: injected RetryOOM
E                               at ai.rapids.cudf.ColumnView.hasNonEmptyNulls(Native Method)
E                               at ai.rapids.cudf.ColumnView.hasNonEmptyNulls(ColumnView.java:5218)
E                               at ai.rapids.cudf.AssertEmptyNulls.assertNullsAreEmpty(AssertEmptyNulls.java:33)
E                               at ai.rapids.cudf.ColumnView.<init>(ColumnView.java:58)
...
```
This bug needs to be tracked down, either in `libcudf` or in `spark-rapids-jni` and sorted out, for `subquery_test.py` tests to run cleanly.

This is not specific to Spark 4. It seems to turn up when ANSI mode is enabled. I'm able to repro this on Spark 3.5, for instance:
```
PYSP_TEST_spark_sql_ansi_enabled=true  SPARK_HOME=/home/mithunr/workspace/dev/spark/bin/spark-3.5.0-bin-hadoop3 TESTS=subquery_test.py::test_scalar_subquery_array ./integration_tests/run_pyspark_from_build.sh
```
Datagen seed: `DATAGEN_SEED=1719427606`.

**Comment by mattahrens** [comment]
Is there a datagen seed needed to reproduce this issue?  

And is this issue specific to Spark 4.0?  Or would any Spark version with ANSI enabled encounter this issue?

**Comment by mythrocks** [comment]
@mattahrens, @SurajAralihalli: Sorry, I should have included that as part of the initial report.
This is not specific to Spark 4. It seems to turn up when ANSI mode is enabled. I'm able to repro this on Spark 3.5, for instance:
```
PYSP_TEST_spark_sql_ansi_enabled=true  SPARK_HOME=/home/mithunr/workspace/dev/spark/bin/spark-3.5.0-bin-hadoop3 TESTS=subquery_test.py::test_scalar_subquery_array ./integration_tests/run_pyspark_from_build.sh
```
Datagen seed: `DATAGEN_SEED=1719427606`.

**Comment by mythrocks** [comment]
@SurajAralihalli and I tried to repro this a couple of days ago, to no avail.

I'm going to try another repro over the next couple of days. If it can't be reproduced, maybe we should close it for now.  We can reopen when this occurs again.

**Comment by mythrocks** [comment]
I spent a fair bit of time last night to try repro this.  It looked like the other fixes (including #11029) were masking the problem, but rolling them back doesn't seem to help in the repro. 

I'm closing this for now.  I'll keep an eye open for whether this happens again.  I'm fairly confident that this hasn't been fixed yet.
